company.py

```diff
@@ -0,0 +1,182 @@
+Третий уровень:
+Теперь вам пригодится ещё список taxes, в котором хранится информация о налогах на сотрудников из разных департаметов.
+Если department None, значит, этот налог применяется ко всем сотрудникам компании.
+Иначе он применяется только к сотрудникам департмента, название которого совпадает с тем, что записано по ключу department.
+К одному сотруднику может применяться несколько налогов.
+13. Вывести список отделов со средним налогом на сотрудников этого отдела.
+14. Вывести список всех сотредников с указанием зарплаты "на руки" и зарплаты с учётом налогов.
+15. Вывести список отделов, отсортированный по месячной налоговой нагрузки.
+16. Вывести всех сотрудников, за которых компания платит больше 100к налогов в год.
+17. Вывести имя и фамилию сотрудника, за которого компания платит меньше всего налогов.
+"""
+
+departments = [
+    {
+        "title": "HR department",
+        "employers": [
+            {"first_name": "Daniel", "last_name": "Berger", "position": "Junior HR", "salary_rub": 50000},
+            {"first_name": "Michelle", "last_name": "Frey", "position": "Middle HR", "salary_rub": 75000},
+            {"first_name": "Kevin", "last_name": "Jimenez", "position": "Middle HR", "salary_rub": 70000},
+            {"first_name": "Nicole", "last_name": "Riley", "position": "HRD", "salary_rub": 120000},
+        ]
+    },
+    {
+        "title": "IT department",
+        "employers": [
+            {"first_name": "Christina", "last_name": "Walker", "position": "Python dev", "salary_rub": 80000},
+            {"first_name": "Michelle", "last_name": "Gilbert", "position": "JS dev", "salary_rub": 85000},
+            {"first_name": "Caitlin", "last_name": "Bradley", "position": "Teamlead", "salary_rub": 950000},
+            {"first_name": "Brian", "last_name": "Hartman", "position": "CTO", "salary_rub": 130000},
+        ]
+    },
+]
+
+taxes = [
+    {"department": None, "name": "vat", "value_percents": 13},
+    {"department": "IT Department", "name": "hiring", "value_percents": 6},
+    {"department": "BizDev Department", "name": "sales", "value_percents": 20},
+]
+
+
+def last_name_vowel(departments):
+    last_name_vowel_final = set()
+    for department in departments:
+        for employers_info in department['employers']:
+            if employers_info['last_name'][-1] in 'a, e, i, o, u, y':
+                last_name_vowel_final.add(employers_info['first_name'])
+    return last_name_vowel_final
+
+def salary_women(departments):
+    salary_women_dep = []
+    for departmen in departments:
+        dep = ''
+        count_women = 0
+        salary_women = 0
+        for employers_info in departmen['employers']:
+            if employers_info['first_name'] in "Michelle, Nicole":
+                count_women += 1
+                salary_women += employers_info['salary_rub']
+                dep = departmen['title']
+        salary_women_dep.append((dep, int(salary_women/count_women) ))
```

Comment:
```suggestion
        if dep is not None:
            salary_women_dep.append((dep, int(salary_women/count_women) ))
        # или даже так
        if count_women > 0:
            salary_women_dep.append((dep, int(salary_women/count_women) ))
```
иначе можно поделить на 0